### ISSUE TITLE
Key-Bindings & Make

### ISSUE THREAD

**Issue by caltrop-3** [geany/geany#1614]
can not take screen-shot while menu open or toolbar open
ie. ALL system key interaction is suspended
but wait, there is a screenshot, taken with delay

not all menu items have key-bindings

did I mention that key-bindings do not update until Geany is restarted

'Build Toolbar' does not match the edited 'Build Menu'

<Primary> ??? same as the control key but why <Primary> ?

'Make Custon Target' even when edited asks for user input; that's OK, I just used the empty line at the bottom but no key-binding available

Problem: I am wanting custom make commands available to ALL my projects. I have a set of make commands that are the same for ALL of my projects.

No I do not use the default name for make. I use my project 'prefix.mak'. Much easier to keep track of the makefiles instead of the default - 'makefile'.
example:
Test.mak
Test.h
Test.c
Test_1.c
Test_fx.c

my make commands:
F8   Make (Make All)       make -f %e.mak
F9   Clean (Make Object)   make -f %e.mak clean
??   Real Clean (empty)    make -f %e.mak realclean
F5   Execute
S-F8 Compile
S-F9 Build

otherwise LOVING IT! reminds me of TurboC I used in DOS way back...

screen shots:
![geany make](https://user-images.githubusercontent.com/31071746/30873599-c94ec648-a2bb-11e7-9af4-4668197943cc.png)
![geany build menu](https://user-images.githubusercontent.com/31071746/30873644-e9baf1fe-a2bb-11e7-881d-51132173b8af.png)
![geany build toolbar](https://user-images.githubusercontent.com/31071746/30873689-18d3ddc0-a2bc-11e7-8034-47351d4fc4c0.png)

**Comment by codebrainz** [comment]
Could you please clearly specify the following:

  - A short summary describing a single bug
  - What is the expected behaviour?
  - What is the actual behaviour?
  - What are the specific steps required to reproduce the bug?

Thank you.

**Comment by elextr** [comment]
As @codebrainz said please provide information to allow bugs to be reproduced or feature requests to be compared against existing capability or otherwise understood.  Also please keep one bug/feature request per issue, omnibus issues are unmanageable and will be closed by github as soon as one item is addressed. Please also provide your OS, the version of Geany, and GTK and Glib (from `Help->Debug messages`) on every issue.

To answer some points that I think I understand (to save you effort of re-entering them as proper issues):

The behaviour of menus closing on loss of focus is normal GTK behaviour in the toolkit, and the delay on screenshots is the correct solution.  This will not change unless GTK changes.  

Not all menu items have _DEFAULT_ keybindings, this is intended, if there are items missing in the keybindings preference please raise a separate issue _listing_ those you found.

`Make Custom Target` has a default keybinding here, `Ctrl+Shift+F9`.

Lack of toolbar items and keybindings for configurable build menu items is a known issue, don't repeat, it will just be closed as duplicate.

The Geany manual tells you how to set build commands for all projects (set them in user and not in projects) http://www.geany.org/manual/current/index.html#build-menu-configuration and its expanded on in the wiki https://wiki.geany.org/howtos/configurebuildmenu.

**Comment by b4n** [comment]
> Lack of toolbar items […] is a known issue

Yes, but I didn't know the tool items labels were not updated according to the customized label, which is kind of confusing.

**Comment by elextr** [comment]
@b4n the toolbar build menu is made once in `build_init()` and never updated.  IIRC doing anything with the toolbar was delayed because the build system changes were already very big and needed releasing for testing.  And "nobody" :tm: has felt the need to go back and do it.

**Comment by ntrel** [comment]
Renamed issue.

> not all menu items have key-bindings

This is covered by #705.

> did I mention that key-bindings do not update until Geany is restarted

Not sure what this means unless talking about changing number of items in each group, which is a various pref.
